lib/new_relic/agent/instrumentation/aws_sqs/instrumentation.rb

```diff
@@ -0,0 +1,89 @@
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+# frozen_string_literal: true
+
+module NewRelic::Agent::Instrumentation
+  module AwsSqs
+    MESSAGING_LIBRARY = 'SQS'
+
+    def send_message_with_new_relic(*args)
+      segment = nil
+      begin
+        info = get_url_info(args[0])
+        segment = NewRelic::Agent::Tracer.start_message_broker_segment(
+          action: :produce,
+          library: MESSAGING_LIBRARY,
+          destination_type: :queue,
+          destination_name: info[:queue_name]
+        )
+        add_aws_attributes(segment, info)
+      rescue => e
+        NewRelic::Agent.logger.error('Error starting message broker segment in Aws::SQS::Client#send_message ', e)
+      end
+      NewRelic::Agent::Tracer.capture_segment_error(segment) do
+        yield
+      end
+    ensure
+      segment&.finish
+    end
+
+    def send_message_batch_with_new_relic(*args)
+      segment = nil
+      begin
+        info = get_url_info(args[0])
+        segment = NewRelic::Agent::Tracer.start_message_broker_segment(
+          action: :produce,
+          library: MESSAGING_LIBRARY,
+          destination_type: :queue,
+          destination_name: info[:queue_name]
+        )
+        add_aws_attributes(segment, info)
+      rescue => e
+        NewRelic::Agent.logger.error('Error starting message broker segment in Aws::SQS::Client#send_message_batch ', e)
+      end
+      NewRelic::Agent::Tracer.capture_segment_error(segment) do
+        yield
+      end
+    ensure
+      segment&.finish
+    end
+
+    def receive_message_with_new_relic(*args)
+      segment = nil
+      begin
+        info = get_url_info(args[0])
+        segment = NewRelic::Agent::Tracer.start_message_broker_segment(
+          action: :consume,
+          library: MESSAGING_LIBRARY,
+          destination_type: :queue,
+          destination_name: info[:queue_name]
+        )
+        add_aws_attributes(segment, info)
+      rescue => e
+        NewRelic::Agent.logger.error('Error starting message broker segment in Aws::SQS::Client#receive_message ', e)
+      end
+      NewRelic::Agent::Tracer.capture_segment_error(segment) do
+        yield
+      end
+    ensure
+      segment&.finish
+    end
+
```

Comment:
What do you think about making the two methods below this line private?

Reply:
https://github.com/newrelic/newrelic-ruby-agent/pull/2679/commits/df6d3ca0acb102e7b55e2c0b89ed32c4a46aabdc